resources/splunk_app.rb

```diff
@@ -136,3 +148,41 @@ def config_scope
     apply_config
   end
 end
+
+class PackagedApp < SplunkApp
+  property :source_url, String, required: true
+
+  resource_name :splunk_app_package
+
+  def package_path
+    return @package_path if @package_path
+    file_cache = Pathname.new(Chef::Config['file_cache_path'])
+    @package_path = file_cache.join(CernerSplunk::PathHelpers.filename_from_url(source_url).gsub(/.spl$/, '.tgz'))
+  end
+
+  action :install do
+    return unless !version || changed?(:version)
+
+    remote_file package_path.to_s do
```

Comment:
@alecxvs Extracting to a temp directory before installing is probably good (assuming you don't just lift the Tarball class from cerner_splunk).

But with the handling of local, I think the default behavior should be to fail on local content being present, with a parameter that if set could say "Yes, delete local content" or "No, keep local content" 

We should indeed be validating that the package version matches the expected version as _one of_ the checks. In cerner_splunk this is handled: https://github.com/cerner/cerner_splunk/blob/2.9.0/libraries/splunk_app.rb#L231-L233 
If the expected version does not match the version from the tarball exactly, then we fail unless we specified to expect a release version and the tarball contains a prerelease of the specified release version.

(We later handle the case that the release version is installed and we're trying to install a prerelease on top). Basically all of the guard conditions to make sure we don't have pain later